### PR TITLE
Improves Files.App.Storage

### PR DESCRIPTION
**Pull request by cinqmilleans**
**Resolved / Related Issues**
Clean Storage code.

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

### CODE REVIEW

**Review by QuaintMako — APPROVED** [review]
LGTM. I haven't seen any problem.

`src/Files.App.Storage/FtpStorage/FtpStorageFolder.cs` and `src/Files.App.Storage/WindowsStorage/WindowsStorageFolder.cs` are alike on some spaces, but I do not think it'd be wise to try to fuse them.

File: src/Files.App.Storage/FtpStorage/FtpFileSystemService.cs [hunk 1]
```diff
@@ -10,16 +10,16 @@ namespace Files.App.Storage.FtpStorage
 {
 	public sealed class FtpFileSystemService : IFileSystemService
 	{
+		/// <inheritdoc/>
 		public Task<bool> IsFileSystemAccessibleAsync(CancellationToken cancellationToken = default)
-		{
```
Comment (d2dyno1) [inline src/Files.App.Storage/FtpStorage/FtpFileSystemService.cs:14]: I'd keep it as is.

File: src/Files.App.Storage/FtpStorage/FtpFileSystemService.cs [hunk 2]
```diff
@@ -41,30 +42,37 @@ public async Task<bool> DirectoryExistsAsync(string path, CancellationToken canc
...
+			var item = await client.GetObjectInfo(ftpPath, token: cancellationToken);
+			if (item is null || item.Type is not FtpObjectType.File)
 				throw new FileNotFoundException("File was not found from path.");
 
 			return new FtpStorageFile(ftpPath, item.Name);
 		}
+
+		private static async Task<AsyncFtpClient> GetClient(string path, CancellationToken cancellationToken = default)
+		{
+			using AsyncFtpClient client = path.GetFtpClient();
```
Comment (d2dyno1) [inline src/Files.App.Storage/FtpStorage/FtpFileSystemService.cs:73]: `using` statement here will dispose the client after the method exits. Remove it from here

File: src/Files.App.Storage/FtpStorage/FtpHelpers.cs [hunk 3]
```diff
@@ -8,59 +8,60 @@ namespace Files.App.Storage.FtpStorage
...
+			var host = path.GetFtpHost();
+			var port = path.GetFtpPort();
+			var credentials = FtpManager.Credentials.Get(host, FtpManager.Anonymous);
 
-			return hostIndex == -1 ? "/" : path.Substring(hostIndex);
+			return new(host, credentials, port);
 		}
 
-		public static Task EnsureConnectedAsync(this AsyncFtpClient ftpClient, CancellationToken cancellationToken = default)
+		public static string GetFtpPath(this string path)
```
Comment (d2dyno1) [inline src/Files.App.Storage/FtpStorage/FtpHelpers.cs:20]: Don't make GetFtpPath an extension method on string!!!!

File: src/Files.Shared/Helpers/PathHelpers.cs [hunk 4]
```diff
@@ -5,7 +5,7 @@ namespace Files.Shared.Helpers
 {
 	public static class PathHelpers
 	{
-		public static string GetParentDir(string path)
+		public static string GetParentDir(this string path)
```
Comment (d2dyno1) [inline src/Files.Shared/Helpers/PathHelpers.cs:8]: Nooo!!!

File: src/Files.Shared/Helpers/PathHelpers.cs [hunk 5]
```diff
@@ -14,7 +14,7 @@ public static string GetParentDir(string path)
 			return path.Substring(0, index != -1 ? index : path.Length);
 		}
 
-		public static string Combine(string folder, string name)
+		public static string Combine(this string folder, string name)
```
Comment (d2dyno1) [inline src/Files.Shared/Helpers/PathHelpers.cs:17]: "Nooo!!!" here too

File: src/Files.App.Storage/FtpStorage/FtpFileSystemService.cs (hunk 2 above)
Comment (d2dyno1) [reply src/Files.App.Storage/FtpStorage/FtpFileSystemService.cs:73]: Besides, you could refactor the method in FtpStorable.GetFtpClientAsync() - to use a `string path` parameter, and remove this method

File: src/Files.App.Storage/FtpStorage/FtpStorable.cs [hunk 6]
```diff
@@ -5,32 +5,32 @@
...
 			Name = name;
-			Id = string.Empty;
+			Path = path.GetFtpPath();
 		}
 
 		public virtual Task<ILocatableFolder?> GetParentAsync(CancellationToken cancellationToken = default)
-		{
-			return Task.FromResult<ILocatableFolder?>(null);
-		}
+			=> Task.FromResult<ILocatableFolder?>(null);
```
Comment (d2dyno1) [inline src/Files.App.Storage/FtpStorage/FtpStorable.cs:27]: Revert this change please :)

File: src/Files.App.Storage/FtpStorage/FtpStorable.cs [hunk 7]
```diff
@@ -5,32 +5,32 @@
 
 namespace Files.App.Storage.FtpStorage
 {
+	/// <inheritdoc cref="IStorable"/>
 	public abstract class FtpStorable : ILocatableStorable
 	{
 		/// <inheritdoc/>
-		public string Path { get; protected set; }
+		public string Id => string.Empty;
```
Comment (d2dyno1) [inline src/Files.App.Storage/FtpStorage/FtpStorable.cs:12]: fyi, I wouldn't do anything to this for now, as `Id` will be implemented [_**here**_](https://github.com/files-community/Files/pull/10477/commits/fcc9db77ba895cbe503b569d521bb73ab8a753d6). (And in other places as well)

File: src/Files.App.Storage/FtpStorage/FtpStorable.cs [hunk 8]
```diff
@@ -5,32 +5,32 @@
...
 		}
 
 		public virtual Task<ILocatableFolder?> GetParentAsync(CancellationToken cancellationToken = default)
-		{
-			return Task.FromResult<ILocatableFolder?>(null);
-		}
+			=> Task.FromResult<ILocatableFolder?>(null);
 
-		protected AsyncFtpClient GetFtpClient()
+		protected async Task<AsyncFtpClient> GetFtpClient(CancellationToken cancellationToken = default)
```
Comment (d2dyno1) [inline src/Files.App.Storage/FtpStorage/FtpStorable.cs:29]: Rename to `GetFtpClientAsync`

File: src/Files.App.Storage/FtpStorage/FtpStorageFile.cs [hunk 9]
```diff
@@ -7,31 +7,23 @@
...
 namespace Files.App.Storage.FtpStorage
 {
+	/// <inheritdoc cref="IFile"/>
 	public sealed class FtpStorageFile : FtpStorable, IModifiableFile, ILocatableFile
 	{
-		public FtpStorageFile(string path, string name)
-			: base(path, name)
-		{
-		}
+		public FtpStorageFile(string path, string name) : base(path, name) {}
```
Comment (d2dyno1) [inline src/Files.App.Storage/FtpStorage/FtpStorageFile.cs:13]: Revert this change :) (And in other places too)

**Review by d2dyno1 — CHANGES_REQUESTED** [review]

**Comment by cinqmilleans** [comment]
@d2dyno1 @yair100 @QuaintMako This pr has a lot of cancellation requests and conflicts with #10477. I am closing it because it is no longer needed.

**Pull request closed without merging**